core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -240,11 +247,25 @@ private[kafka] object LogValidator extends Logging {
     for (batch <- records.batches.asScala) {
       validateBatch(topicPartition, firstBatch, batch, isFromClient, magic, brokerTopicStats)
 
+      val batchOffsetCounter = new LongRef(0)
+
       var maxBatchTimestamp = RecordBatch.NO_TIMESTAMP
       var offsetOfMaxBatchTimestamp = -1L
 
       for (record <- batch.asScala) {
-        validateRecord(batch, topicPartition, record, now, timestampType, timestampDiffMaxMs, compactedTopic, brokerTopicStats)
+        val batchIndex = batchOffsetCounter.getAndIncrement()
```

Comment:
It would be nice to be consistent on terminology. If we are using `relativeOffset` elsewhere, let's do the same here. I prefer `batchIndex` because there is little chance of confusion with absolute offsets. If you agree, let's update the rest of the code to prefer this term as well.

Reply:
ack -- I changed `relativeOffset` to `batchIndex` in other places also